mlflow/data_mlflow_model.go

```diff
@@ -0,0 +1,45 @@
+package mlflow
+
+import (
+	"context"
+	"fmt"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/ml"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceModel() *schema.Resource {
+	return common.WorkspaceData(func(ctx context.Context, data *struct {
+		Name         string            `json:"name,omitempty" tf:"computed"`
+		Version      string            `json:"version,omitempty" tf:"computed"`
+		ModelVersion []ml.ModelVersion `json:"model_versions,omitempty" tf:"computed"`
```

Comment:
This only returns a single model. Why not use *ml.ModelVersion?

Reply:
A model can have an array of model versions. (Ref: [document](https://docs.databricks.com/api/workspace/modelregistry/getmodel))

If no version is specified, all latest version models are returned. If a model version is specified, it will only return that version.

I think it is more user friendly if it stays as an array even if only one model is returned to avoid confusion. But if you think there is a better approach to this, please let me know.